functional/FunctionalTest.java

```diff
@@ -1103,14 +1097,79 @@ public static void statObject_test3() throws Exception {
       }
     } catch (Exception e) {
       mintFailedLog(
-          "statObject(String bucketName, \"randomName/\")",
+          "statObject(StatObjectArgs args) with non-existing object",
           null,
           startTime,
           null,
           e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
       throw e;
     } finally {
-      mintSuccessLog("statObject(String bucketName, \"randomName/\"`)", null, startTime);
+      mintSuccessLog("statObject(StatObjectArgs args) with non-existing object", null, startTime);
+    }
+  }
+
+  /** Test: with extra headers/query params: statObject(StatObjectArgs args). */
+  public static void statObject_test4() throws Exception {
+    if (!mintEnv) {
+      System.out.println("Test: with extra headers/query params: statObject(StatObjectArgs args)");
+    }
+
+    long startTime = System.currentTimeMillis();
+    try {
+      String objectName = getRandomName();
+      Map<String, String> headerMap = new HashMap<>();
+      headerMap.put("Content-Type", customContentType);
+      headerMap.put("my-custom-data", "foo");
+      try (final InputStream is = new ContentInputStream(1)) {
+        PutObjectOptions options = new PutObjectOptions(1, -1);
+        options.setHeaders(headerMap);
+        options.setContentType(customContentType);
+        client.putObject(bucketName, objectName, is, options);
+      }
```

Comment:
The first part inside `try` block can be extracted out into a method, say `putTestObject()`